test/src/test/java/org/neo4j/ogm/autoindex/BaseAutoIndexManagerTestClass.java

```diff
@@ -221,9 +219,7 @@ public void testAutoIndexDumpCreatesIndex() throws IOException {
 
     void runAutoIndex(String mode) {
         Configuration configuration = getBaseConfiguration().autoIndex(mode).build();
-        Neo4jSession session = (Neo4jSession) sessionFactory.openSession();
-        AutoIndexManager indexManager = new AutoIndexManager(metaData, configuration, session);
-        indexManager.build();
+        new AutoIndexManager(metaData, configuration, sessionFactory);
```

Comment:
I understand the concerns and I'd say it's a valid point for a later improvement and should be out-of-scope for this PR, since it will require more work, increase complexity and does not bring value to this particular PR main goal. As I mentioned before, this change was a minor improvement regarding the usage of the sessionFactory.